### ISSUE TITLE
prop-types doesn't complain about destructured props in stateless functions

### ISSUE THREAD

**Issue by kesne** [yannickcr/eslint-plugin-react#354]
The following should cause an error for missing proptype validations, but it doesn't.
```
const Test = ({ name }) => {
  return (
    <div>{name}</div>
  );
};
```

If you replace the destructuring with using the props object, it works as expected.

**Comment by yannickcr** [comment]
Yep, this pattern is not actually handled.

**Comment by ljharb** [comment]
Would love to see this handled, since it's the primary pattern people seem to use for functional stateless components.

**Comment by ljharb** [comment]
Any progress on this?
